[WebGPU] Sliced BC format in webgpu.github.io/webgpu-samples/?sample=volumeRenderingTexture3D does not work

mwyrzykowski opened the pull request:
#### d15d157635eb35587837e1b91e746d5ba204aeef
<pre>
[WebGPU] Sliced BC format in webgpu.github.io/webgpu-samples/?sample=volumeRenderingTexture3D does not work
<a href="https://bugs.webkit.org/show_bug.cgi?id=291690">https://bugs.webkit.org/show_bug.cgi?id=291690</a>
<a href="https://rdar.apple.com/149489131">rdar://149489131</a>

Reviewed by Tadeu Zagallo.

Samples site added support for 3D compressed textures, which
we already support, but our validation was incorrect.

* Source/WebGPU/WebGPU/Texture.h:
* Source/WebGPU/WebGPU/Texture.mm:
(WebGPU::Texture::isCompressedFormat):
(WebGPU::Texture::compressedFormatType):
(WebGPU::Device::errorValidatingTextureCreation):
Allow ASTC and BC 3D compressed textures with the extension enabled.

Canonical link: <a href="https://commits.webkit.org/294078@main">https://commits.webkit.org/294078@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/ea0ac34936bdd6aed53b6026659e751752dafaa8

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/100749 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/20401 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/10700 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/105886 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/51338 "Built successfully") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/20710 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/28875 "Built successfully") | [❌ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/76716 "Found 2 new test failures: imported/w3c/web-platform-tests/fullscreen/rendering/ua-style-iframe.html imported/w3c/web-platform-tests/html/semantics/forms/constraints/infinite_backtracking.html (failure)") | [✅ 🧪 win-tests](https://ews-build.webkit.org/#/builders/60/builds/33754 "Passed tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/103756 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/15910 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/91006 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/57069 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/15719 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/135/builds/9014 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/50713 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/85622 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/9089 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/108241 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/27867 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/20470 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/85675 "Passed tests") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/28230 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/87208 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/85217 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/21689 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/29924 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/7656 "Passed tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/21871 "Built successfully") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/27802 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/33057 "Built successfully") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/27613 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/30931 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/29171 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

mwyrzykowski commented on Source/WebGPU/WebGPU/Texture.mm:
```diff
@@ -68,6 +74,7 @@
     case WGPUTextureFormat_EACR11Snorm:
     case WGPUTextureFormat_EACRG11Unorm:
     case WGPUTextureFormat_EACRG11Snorm:
+        return Texture::CompressFormat::ETC;
```
`EAC` formats are considered `ETC`

tadeuzagallo reviewed (approved):
I understand the issue was caught in the sample website, but is this covered by the CTS or do we need to add a test?

mwyrzykowski committed:
[WebGPU] Sliced BC format in webgpu.github.io/webgpu-samples/?sample=volumeRenderingTexture3D does not work
https://bugs.webkit.org/show_bug.cgi?id=291690
rdar://149489131

Reviewed by Tadeu Zagallo.

Samples site added support for 3D compressed textures, which
we already support, but our validation was incorrect.

* Source/WebGPU/WebGPU/Texture.h:
* Source/WebGPU/WebGPU/Texture.mm:
(WebGPU::Texture::isCompressedFormat):
(WebGPU::Texture::compressedFormatType):
(WebGPU::Device::errorValidatingTextureCreation):
Allow ASTC and BC 3D compressed textures with the extension enabled.

Canonical link: https://commits.webkit.org/294078@main
